lib/Backend/BackwardPass.cpp

```diff
@@ -7115,7 +7119,7 @@ BackwardPass::ProcessBailOnNoProfile(IR::Instr *instr, BasicBlock *block)
     curInstr = instr->m_prev;
 
     // Move to top of block (but just below any fgpeeptemp lds).
-    while(!curInstr->StartsBasicBlock() && curInstr != instrNope)
+    while(!curInstr->StartsBasicBlock() && curInstr->m_opcode != Js::OpCode::Catch && curInstr->m_opcode != Js::OpCode::Finally && curInstr != instrNope)
```

Comment:
>>m_opcode != Js::OpCode::Catch && curInstr->m_opcode != Js::OpCode::Finally && curInst [](start = 52, length = 86)

This seems redundant with the check at the top of the function.

Reply:
Yes! Changed it.